examples/AffinePlaneCurve.jl

```diff
@@ -0,0 +1,124 @@
+module AffinePlaneCurveModule
+
+using Oscar, Markdown
+
+export AffinePlaneCurve, AffinePlaneCurve_degree, AffinePlaneCurve_Jacobian, AffinePlaneCurve_tangent, AffinePlaneCurve_issmoothpoint, AffinePlaneCurve_union
+
+####################################################
+#
+#We follow the lecture notes by Andreas Gathmann: https://www.mathematik.uni-kl.de/~gathmann/en/curves.php
+#
+####################################################
+mutable struct AffinePlaneCurve{S}
+  eq::Oscar.MPolyElem                      #Equation of the curve (polynomial in two variables)
+  degree::Union{String, Int64}             #degree of the equation of the curve
+  dimension::Int64                         #dimension of the curve (as a variety)
+  function AffinePlaneCurve(eq::Oscar.MPolyElem{T}) where {T <: FieldElem}
+    r = new{T}()
+    r.eq = eq
+    if nvars(parent(r.eq))!=2              #If the ambiant ring does not have two variables, show an error.
+       error("The defining equation must belong to a ring with two variables")
+    else
+       r.degree = "not computed yet"
```

Comment:
If this is the only reason to allow a string for the degree then you may consider using a Nullable type instead. See the documentation here and search for the word "Nullable"

https://docs.julialang.org/en/v1/manual/types/